sdk/webpubsub/Azure.Messaging.WebPubSub/src/WebPubSubServiceClient_extensions.cs

```diff
@@ -70,36 +118,11 @@ public WebPubSubServiceClient(Uri endpoint, string hub, AzureKeyCredential crede
             Pipeline = HttpPipelineBuilder.Build(
                 options,
                 perCallPolicies: perCallPolicies,
-                perRetryPolicies: new HttpPipelinePolicy[] { new WebPubSubAuthenticationPolicy(credential) },
+                perRetryPolicies: new HttpPipelinePolicy[] { new WebPubSubAuthenticationPolicy(endpoint) },
```

Comment:
I recommend using `BearerTokenAuthenticationPolicy` for `TokenCredential`. It supports many features like token caching and background refresh.